### PR TITLE
Add `Timeline` component

### LINKED ISSUE #275: Timeline component
image

### PR DESCRIPTION
**Pull request by hrach**
closes #275

TODO:
- [x] merge colors & rebase
- [x] consider finishing autoscaling PR (#86) and then fix the sizing here

https://user-images.githubusercontent.com/284263/198893252-cd88368c-4204-4a18-b371-7cc00aaa559e.mp4

### CODE REVIEW

File: ui/src/main/java/kiwi/orbit/compose/ui/controls/Timeline.kt [hunk 1]
```diff
@@ -0,0 +1,416 @@
...
+ *
+ * ```
+ * Timeline {
+ *     TimelineItem(TimelineItemStatus.PastSuccess) { Text("Step 1") }
+ *     TimelineItem(TimelineItemStatus.InProgress) { Text("Step 2") }
+ *     TimelineItem(TimelineItemStatus.Future) { Text("Step 3") }
+ * }
+ * ```
+ *
+ * The current "active" state is modelled by Current* or InProgress states.
```
Comment (Drekorian) [inline ui/src/main/java/kiwi/orbit/compose/ui/controls/Timeline.kt:68]: US English please, _modeled_.

Also I don't see _Current*_ states?

File: ui/src/main/java/kiwi/orbit/compose/ui/controls/Timeline.kt [hunk 2]
```diff
@@ -0,0 +1,416 @@
...
+         * 1. item 0
+         * 3. line for item 0
+         * 2. line gradient above item 0
+         * 4. item 1
+         * 6. line for item 1
+         * 5. line gradient above item 1
+         * ...
+         */
+        check(measurables.size.mod(3) == 0) {
+            "Timeline container can render only TimelineItem composables."
```
Comment (Drekorian) [inline ui/src/main/java/kiwi/orbit/compose/ui/controls/Timeline.kt:90]: I don't see the connection between the `check` and the error message.

**Review by Drekorian — APPROVED** [review]

File: ui/src/main/java/kiwi/orbit/compose/ui/controls/Timeline.kt (hunk 2 above)
Comment (hrach) [reply ui/src/main/java/kiwi/orbit/compose/ui/controls/Timeline.kt:90]: Every item has to render "three" composables. So when there is no correct composables count, it means you're not using TimelineItem. Yes, a lot of other cases are not covered (not using it and having 3 composables), but this is some fallback not to throw.

File: ui/src/main/java/kiwi/orbit/compose/ui/controls/Timeline.kt (hunk 1 above)
Comment (hrach) [reply ui/src/main/java/kiwi/orbit/compose/ui/controls/Timeline.kt]: thx, fixed, renamed back to current

File: ui/src/main/java/kiwi/orbit/compose/ui/controls/Timeline.kt (hunk 2 above)
Comment (Drekorian) [reply ui/src/main/java/kiwi/orbit/compose/ui/controls/Timeline.kt:90]: I see, something along the lines of `"Invalid number of composables: ${measurables.size}"` would probably be better.

**Pull request merged**